hack/istio/multicluster/deploy-kiali.sh

```diff
@@ -121,17 +157,22 @@ deploy_kiali() {
     --set health_config.rate[0].tolerance[0].code="5xx"
     --set health_config.rate[0].tolerance[0].degraded=2
     --set health_config.rate[0].tolerance[0].failure=100
-    --set deployment.ingress.enabled="false"
-    --set deployment.service_type="LoadBalancer"
-    --set server.web_port="80"
-    --repo https://kiali.org/helm-charts
+    --set deployment.ingress.enabled=${ingress_enabled}
+    --set deployment.service_type=${service_type}
+    --set server.web_port=${web_port}
     kiali-server
     ${KIALI_SERVER_HELM_CHARTS}'
 
   eval $helm_command
+  if [ "${KIALI_AUTH_STRATEGY}" == "openshift" ]; then
+    local kiali_route_url
+    kiali_route_url=$(kubectl get route kiali -n "${ISTIO_NAMESPACE}" -o jsonpath='{.spec.host}')
+    ${CLIENT_EXE} get oauthclients --context "${CLUSTER1_CONTEXT}" -n "${ISTIO_NAMESPACE}" kiali-"${ISTIO_NAMESPACE}" -o json | jq ".redirectURIs = [\"https://${kiali_route_url}/api/auth/callback/${CLUSTER2_CONTEXT}\"]" | ${CLIENT_EXE} --force=true --context "${CLUSTER2_CONTEXT}" apply -f -
+  else
   # Helm chart doesn't support passing in service opts so patch them after the helm deploy.
```

Comment:
Is this useful? This helm value does set the service port: https://github.com/kiali/helm-charts/blob/v1.84.0/kiali-server/templates/service.yaml#L36

Reply:
What this is doing is creating the `OAuthClient` on the _remote_ cluster which the helm chart can't do since it only deals with the cluster kiali is deployed onto. There needs to be an `OAuthClient` on each openshift cluster since that's what registers the kiali app as an oauth client of that cluster's oauth server. This requirement needs to be documented somewhere and it might be worth providing a script that does this or else modifying `kiali-prepare-remote-cluster` to also do this when the target cluster is openshift.

Reply:
Yeah, we should create a github issue on that so we remember to do this. Or, make that change in this PR. Up to you if you want to implement that in this PR or defer that and create a github issue so we don't forget